syntax = "proto2";

package st;

import "modules/cnoa_pnc/planning/proto/lane_point.proto";

message RouteSectionSequenceProto {
  optional double start_fraction = 1;
  optional double end_fraction = 2;
  repeated uint64 section_id = 3;
  optional mapping.LanePointProto destination = 4;
}

message RouteLaneInfoProto {
  optional uint64 lane_id = 1;
  optional double max_driving_distance = 2;
  optional double max_driving_distance_before_lc = 3;
  optional double recommend_driving_distance = 4;
  optional int32 min_lc_num_to_target = 5;
  optional int32 lc_num_within_driving_dist = 6;
  optional double len_before_merge_lane = 7;
}

message NaviSectionInfoProto {
  enum Direction {
    STRAIGHT = 1;
    LEFT_TURN = 2;
    RIGHT_TURN = 3;
  }

  optional uint64 section_id = 1;
  optional double length_before_intersection = 2;
  optional Direction intersection_direction = 3;
}

message RouteNaviInfoProto {
  repeated RouteLaneInfoProto route_lane_info = 1;
  optional RouteSectionSequenceProto back_extend_sections = 2;
  repeated NaviSectionInfoProto navi_section_info = 3;
}
